Orange/widgets/data/oweditdomain.py

```diff
@@ -514,10 +522,15 @@ def _on_variable_changed(self):
 
         # Replace the variable in the 'Domain Features' view/model
         old_var = self.input_vars[self.selected_index]
-        new_var = editor.get_data().copy(compute_value=Orange.preprocess.transformation.Identity(old_var))
+        new_var = editor.get_data().copy(
+            compute_value=Orange.preprocess.transformation.Identity(old_var)
+        )
         self.domain_model[self.selected_index] = new_var
-
-
+        var_names = [vn.name for vn in self.domain_model]
```

Comment:
This is now checked only on variable changes. If I see the warning and reset a variable to the original, the warning does not disappear (the reset functions don't reach this code).
Maybe the check could be moved to commit:
- if everything is ok, remove the error and proceed
- if not, show the error and send None